src/hotspot/share/gc/shenandoah/shenandoahGeneration.cpp

```diff
@@ -791,40 +779,40 @@ void ShenandoahGeneration::prepare_regions_and_collection_set(bool concurrent) {
       ShenandoahFinalMarkUpdateRegionStateClosure old_cl(nullptr);
       heap->old_generation()->parallel_heap_region_iterate(&old_cl);
     }
-
-    heap->assert_pinned_region_status();
```

Comment:
This is called again immediately inside collection set selection so I removed this invocation.